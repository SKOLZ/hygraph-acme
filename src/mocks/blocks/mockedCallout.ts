export const mockedCallout = {
  __typename: "Callout" as const,
  id: "clt7bqmp49dkq07uj23atb9h5",
  title: "Start your 30-day free trial",
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "Join over 4,000+ startups already growing with us.",
            },
          ],
        },
      ],
    },
  },
  actionButton: {
    buttonText: "Get started",
    size: null,
  },
};
