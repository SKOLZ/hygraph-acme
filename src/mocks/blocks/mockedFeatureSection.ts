import { BlockFeatureSection, Direction, Sizes } from "@/generated/graphql";

export const mockedFeatureSection = {
  __typename: "FeatureSection" as const,
  id: "clt7bqmp09dkf07uj2k6in82a",
  label: "feature section label",
  title: "feature section title",
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "As a composable and headless platform, Acme works with all your preferred frontend frameworks, existing systems, GraphQL and REST APIs.",
            },
          ],
        },
      ],
    },
  },
  media: {
    __typename: "StaticImage" as const,
    image: {
      url: "https://media.graphassets.com/bbd3PKNcTBGVkNbcois7",
      width: 480,
      height: 320,
    },
    altText: "Stats page showing a profile views chart and a chat window",
  },
  direction: Direction.LeftToRight,
  featureSectionButton: {
    buttonText: "Get started",
    size: Sizes.Lg,
  },
};
