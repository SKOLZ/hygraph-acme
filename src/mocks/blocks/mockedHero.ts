import { BlockHero, Sizes } from "@/generated/graphql";

export const mockedHero = {
  __typename: "Hero" as const,
  id: "1",
  title: `Hero title`,
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "Hero description",
            },
          ],
        },
      ],
    },
  },
  buttonActions: [
    {
      id: "clt7bqmoz9dkc07ujrs01ghn3",
      buttonText: "Get started",
      size: Sizes.Lg,
    },
  ],
};
