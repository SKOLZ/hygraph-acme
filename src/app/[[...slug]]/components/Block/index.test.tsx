import { render } from "@testing-library/react"
import { mockedHero } from "@/mocks/blocks/mockedHero";
import { mockedFeatureSection } from "@/mocks/blocks/mockedFeatureSection";
import { mockedCallout } from "@/mocks/blocks/mockedCallout";
import { Block } from ".";

describe("Block", () => {
  it("renders the Hero based on the provided type", () => {
    const { container } = render(<Block block={mockedHero} />);
    expect(container).toMatchSnapshot();
  })

  it("renders the FeatureSection based on the provided type", () => {
    const { container } = render(<Block block={mockedFeatureSection} />);
    expect(container).toMatchSnapshot();
  })

  it("renders the Callout based on the provided type", () => {
    const { container } = render(<Block block={mockedCallout} />);
    expect(container).toMatchSnapshot();
  })
})
