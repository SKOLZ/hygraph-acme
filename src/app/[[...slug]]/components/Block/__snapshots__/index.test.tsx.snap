// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders the Callout based on the provided type 1`] = `
<div>
  <div
    class="mx-auto w-full max-w-screen-xl px-8"
  >
    <div
      class="flex flex-col items-start justify-between gap-8 rounded-2xl bg-secondary-800 p-8 xl:flex-row xl:p-16"
    >
      <div
        class="flex flex-col items-start gap-4"
      >
        <h2
          class="title-2 text-white"
        >
          Start your 30-day free trial
        </h2>
        <p
          class="text-1 text-white"
        >
          Join over 4,000+ startups already growing with us.
        </p>
      </div>
      <a
        class="button button-md "
        href="#"
      >
        Get started
      </a>
    </div>
  </div>
</div>
`;

exports[`Block renders the FeatureSection based on the provided type 1`] = `
<div>
  <div
    class="mx-auto flex max-w-screen-xl flex-col gap-12 px-8  lg:flex-row lg:items-center lg:gap-24"
  >
    <div
      class="flex flex-col items-start lg:flex-1"
    >
      <p
        class="text-4"
      >
        feature section label
      </p>
      <h2
        class="title-2 mb-4"
      >
        feature section title
      </h2>
      <p
        class="text-3 text-secondary-600"
      >
        As a composable and headless platform, Acme works with all your preferred frontend frameworks, existing systems, GraphQL and REST APIs.
      </p>
      <a
        class="button button-lg mt-8"
        href="#"
      >
        Get started
      </a>
    </div>
    <div
      class="flex justify-center lg:flex-1 lg:rounded-3xl lg:bg-secondary-100"
    >
      <img
        alt="Stats page showing a profile views chart and a chat window"
        class="rounded border border-gray-50 min-w-0 lg:m-10"
        data-nimg="1"
        decoding="async"
        height="320"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fbbd3PKNcTBGVkNbcois7&w=1080&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fbbd3PKNcTBGVkNbcois7&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fbbd3PKNcTBGVkNbcois7&w=1080&q=75 2x"
        style="color: transparent;"
        width="480"
      />
    </div>
  </div>
</div>
`;

exports[`Block renders the Hero based on the provided type 1`] = `
<div>
  <div
    class="mx-auto w-full flex max-w-screen-xl flex-col items-center px-8 text-center"
  >
    <h1
      class="title-1 mb-4 text-secondary-900 sm:mb-6"
    >
      Hero title
    </h1>
    <p
      class="text-2 mb-8 max-w-3xl text-secondary-400 sm:mb-12"
    >
      Hero description
    </p>
    <div
      class="flex flex-col items-center justify-center gap-3 self-stretch sm:flex-row"
    >
      <a
        class="button button-lg "
        href="#"
      >
        Get started
      </a>
    </div>
  </div>
</div>
`;
